### PR TITLE
feat: minor tweaks to adjust the creation of a prescription object in…

### PR DESCRIPTION
**Pull request by Kavindu-248**
… assesment app

### CODE REVIEW

File: apps/assesment/views.py [hunk 1]
```diff
@@ -38,7 +39,9 @@ def perform_create(self, serializer):
     def perform_update(self, serializer):
         appointment = self.get_object()
         if appointment.status == AppointmentStatus.COMPLETED:
-            raise ValidationError('This appointment is already Completed hence you cannot update it')
+            prescription = Prescription.objects.create(appointment=appointment)
```
Comment (Raam124) [inline apps/assesment/views.py:42]: what if the prescription is not available?

File: apps/assesment/views.py [hunk 2]
```diff
@@ -80,6 +83,10 @@ def perform_update(self, serializer):
             if ongoing_assessments.exists():
                 raise ValidationError('You have a pending assesment.')
             else:
+                form_assesment = self.get_object()
+                if form_assesment.status == 'COMPLETED':
+                    Prescription.objects.create(form_assesment=form_assesment)
```
Comment (Raam124) [inline apps/assesment/views.py:88]: what if the prescription is not available?

File: apps/dispensary/views.py [hunk 3]
```diff
@@ -21,6 +21,10 @@ def retrieve(self, request, pk=None):
         serializer = MedicineSerializer(medicine)
         return Response(serializer.data)
 
+class PrescriptionViewSet(viewsets.ModelViewSet):
```
Comment (Raam124) [inline apps/dispensary/views.py:24]: model view set allows all methods inbuilt we don't need those here

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 1 above)
Comment (Kavindu-248) [reply apps/assesment/views.py:42]: added a validation error.

File: apps/assesment/views.py (hunk 2 above)
Comment (Kavindu-248) [reply apps/assesment/views.py]: added a validation error.

File: apps/dispensary/views.py (hunk 3 above)
Comment (Kavindu-248) [reply apps/dispensary/views.py]: changed to viewset with list and update methods

File: apps/assesment/views.py [hunk 4]
```diff
@@ -38,7 +39,12 @@ def perform_create(self, serializer):
     def perform_update(self, serializer):
         appointment = self.get_object()
         if appointment.status == AppointmentStatus.COMPLETED:
-            raise ValidationError('This appointment is already Completed hence you cannot update it')
+            prescription = Prescription.objects.create(appointment=appointment)
+            if prescription:
+                serializer.save(prescription=prescription)
+                super().perform_update(serializer)
+            else:
```
Comment (Raam124) [inline apps/assesment/views.py:46]: reread the scenario prescription is optional. these are requirements

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 4 above)
Comment (Kavindu-248) [reply apps/assesment/views.py:46]: ammended the code to match if a prescripotion is needed in the payload prescription gets passed as 'true'

File: apps/assesment/views.py [hunk 5]
```diff
@@ -37,8 +38,13 @@ def perform_create(self, serializer):
 
     def perform_update(self, serializer):
         appointment = self.get_object()
+        prescription_needed = self.request.data.get('prescription', False)
```
Comment (Raam124) [inline apps/assesment/views.py:41]: this is not the way to check if a key is there

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py [hunk 6]
```diff
@@ -38,7 +39,11 @@ def perform_create(self, serializer):
     def perform_update(self, serializer):
         appointment = self.get_object()
         if appointment.status == AppointmentStatus.COMPLETED:
-            raise ValidationError('This appointment is already Completed hence you cannot update it')
+            if 'prescription' in self.request.data:
+                prescription = Prescription.objects.create(appointment=appointment)
+                serializer.save(prescription=prescription)
+            else:
+                serializer.save(prescription=None)
```
Comment (Raam124) [inline apps/assesment/views.py:46]: why are you passing none here prescription is not a mandatory field right?

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 6 above)
Comment (Kavindu-248) [reply apps/assesment/views.py]: Yea prescription is optional ,checked by default it will be null, should have not explicitly set it.

**Review by Raam124 — APPROVED** [review]

**Pull request merged**